UI/DamageIndicator.cs

```diff
@@ -0,0 +1,44 @@
+using Godot;
+using System;
+
+public class DamageIndicator : TextureRect
+{
+    // Angle the bullet came from
+    Vector2 ShotFirePosition;
```

Comment:
Vector3

Reply:
I'm keeping this one as a Vector2 as there is no reason to flatten it each frame

Reply:
In that case flatten early and include `Flat` in the variable name